### ISSUE TITLE
[Feature Request] Support for gitea as account, pull request provider and source for notifications

### ISSUE THREAD

**Issue by hypervtechnics** [ForkIssues/TrackerWin#1144]
It would be great to have gitea as a supported/integrated system in Fork.

Meaning support for:

- opening/seeing pull requests
- open in browser
- show notifications
- authentication through the accounts window

The API is described here and should be easy to integrate: 
https://try.gitea.io/api/swagger

**Comment by DanPristupov** [comment]
My question may sound strange, but do you use gitea yourself? Or you'd like Fork to support it?

**Comment by hypervtechnics** [comment]
I don't understand the question directly. I am using Gitea as replacement for Gitlab which is too heavy (the server side). It would be nice that Fork detects by using e.g. the remote url that i am using the gitea server and show some additional information/actions like with GitHub/GitLab/ADO by using the API. Also open in web would be nice. And cloning repositories through the accounts screen.

TLDR: The same as the GitLab self-hosted functionality. Just using gitea in the backend which would probably some adaptions to the API calls.

Did that answer the question? 🤔 I am unsure. If you need more information please ask 😄

**Comment by DefinitelyADev** [comment]
This would be a really nice addition here:

![image](https://user-images.githubusercontent.com/20267678/119123078-e1407e80-ba37-11eb-8d8c-4b195de4b52a.png)

I have set up Gitea for the organization I work for and for my home lab.

**Comment by DefinitelyADev** [comment]
Additionally, it would also be good publicity for fork if it is included in [Awesome Gitea](https://gitea.com/gitea/awesome-gitea).

**Comment by hypervtechnics** [comment]
Thank you for this! 🥳 @DanPristupov

**Comment by DanPristupov** [comment]
Ah, yes. I forgot to mention that here :). You are welcome!

**Comment by jpraet** [comment]
In Gitea, a different domain can be used for SSH vs HTTPS access:

```
[server]
DOMAIN           = git.mycompany.com
SSH_DOMAIN       = git-ssh.mycompany.com
```

In such case, the Gitea integration in Fork currently only works for HTTPS remote urls.
Remotes using the different ssh domain are not recognized.

**Comment by mco-dgmr** [comment]
I like the Gitea account integration, but I ran into two problems:

1. We are using a locally hosted Gitea instance on a non-standard port number. When using the "Create Pull Request on 'origin'" option, the opened URL does not contain the port number, and so the page doesn’t work.
2. Even if I add the non-standard port number manually, the link doesn’t work, because Gitea also expects the target branch in the compare URL. So Fork currently generates the URL `https://scm.example/group/repo/compare/feature%2Fbranch`, whereas it should use `https://scm.example:5000/group/repo/compare/master...feature%2Fbranch` (if the pull request is to be merged into `master`).

We’re using Gitea 1.16.9, btw.

**Comment by hypervtechnics** [comment]
@mco-dgmr I think this should be moved to a new issue.

@DanPristupov From my side I guess this can be closed 👍🏼 

Thanks again for implementing it! ❤️
